salt/states/boto3_route53.py

```diff
@@ -798,3 +826,204 @@ def rr_absent(name, HostedZoneId=None, DomainName=None, PrivateZone=False,
+                    update = True
+                if AlarmIdentifier.get('Region') != check.get('AlarmIdentifier', {}).get('Region'):
+                    update = True
+        for key in resettables:
+            if params.get(key) is None and check.get(key) is not None:
+                update = True
+                reset_elements.append(key)
+
+    if not create and not update:
+        ret['comment'] = 'Route 53 health check {} is already in the desired state.'.format(Name)
+        log.info(ret['comment'])
+        return ret
+    elif create:
+        keys = [
+            'IPAddress', 'Port', 'Type', 'ResourcePath', 'FullyQualifiedDomainName', 'SearchString',
+            'RequestInterval', 'FailureThreshold', 'MeasureLatency', 'Inverted', 'Disabled', 'HealthThreshold',
+            'EnableSNI', 'InsufficientDataHealthStatus', 'Regions', 'AlarmIdentifier', 'ChildHealthChecks', 'CallerReference'
+        ]
+
+        items = {key: params.get(key) for key in keys if params.get(key) is not None}
+        args = {'Name': Name, 'region': region, 'key': key, 'keyid': keyid, 'profile': profile}
+        args.update(items)
+        log.debug(' ==> {}'.format(items))
+
+        if __opts__['test']:
+            ret['comment'] = 'Route 53 health check {} would be created.'.format(Name)
+            ret['changes'] = items
+            ret['result'] = None
+            return ret
+
+        check = __salt__['boto3_route53.create_health_check'](**args)
+        if check:
+            ret['comment'] = 'Route 53 health check {} successfully created.'.format(Name)
+            log.info(ret['comment'])
+            ret['changes']['new'] = check
+        else:
+            ret['comment'] = 'Creation of Route 53 health check {} failed.'.format(Name)
+            log.error(ret['comment'])
+            ret['result'] = False
+            return ret
+    elif update:
+        keys = [
+            'HealthCheckVersion', 'IPAddress', 'Port', 'Type', 'ResourcePath', 'FullyQualifiedDomainName', 'SearchString',
+            'RequestInterval', 'FailureThreshold', 'MeasureLatency', 'Inverted', 'Disabled', 'HealthThreshold',
+            'EnableSNI', 'InsufficientDataHealthStatus', 'Regions', 'AlarmIdentifier', 'ChildHealthChecks', 'CallerReference'
+        ]
+
+        check = checks[0].get('HealthCheckConfig')
+        args = {'region': region, 'key': key, 'keyid': keyid, 'profile': profile}
+        changes = {key: params.get(key) for key in keys if params.get(key) is not None}
+
+        if len(reset_elements) > 0:
+            changes.update({'ResetElements': reset_elements})
+
+        if __opts__['test']:
+            ret['comment'] = 'Route 53 health check {} would be updated.'.format(Name)
+            keys = [key for key in keys if changes.get(key) is not None]
+            ret['changes'] = {key: changes.get(key) for key in keys if _ensure_unicode(changes.get(key)) != _ensure_unicode(check.get(key))}
+            ret['result'] = None
```

Comment:
same here - not sure if this should be `None` or `True`

Reply:
On existing methods like hosted_zone_present, hosted_zone_absent, rr_present, rr_absent, etc. What I saw as pattern was setting result to None when test=true. So I replicated it on this new method.